optimizer_wrapper.rb

```diff
@@ -709,63 +708,71 @@ def self.route_details(vrp, route, vehicle)
     previous = nil
     details = nil
     segments = route[:activities].reverse.collect{ |activity|
-      current = nil
-      if activity[:point_id]
-        current = vrp.points.find{ |point| point[:id] == activity[:point_id] }
-      elsif activity[:service_id]
-        current = vrp.points.find{ |point| point[:id] ==  vrp.services.find{ |service| service[:id] == activity[:service_id] }[:activity][:point_id] }
-      elsif activity[:pickup_shipment_id]
-        current = vrp.points.find{ |point| point[:id] ==  vrp.shipments.find{ |shipment| shipment[:id] == activity[:pickup_shipment_id] }[:pickup][:point_id] }
-      elsif activity[:delivery_shipment_id]
-        current = vrp.points.find{ |point| point[:id] ==  vrp.shipments.find{ |shipment| shipment[:id] == activity[:delivery_shipment_id] }[:delivery][:point_id] }
-      elsif activity[:rest_id]
-        current = previous
-      end
-      segment = if previous && current
-        [current[:location][:lat], current[:location][:lon], previous[:location][:lat], previous[:location][:lon]]
-      end
+      current =
+        if activity[:rest_id]
+          previous
+        else
+          vrp.points.find{ |point| point.id == activity[:point_id] }
+        end
+      segment =
+        if previous && current
+          [current[:location][:lat], current[:location][:lon], previous[:location][:lat], previous[:location][:lon]]
+        end
       previous = current
       segment
     }.reverse.compact
+
     unless segments.empty?
       details = OptimizerWrapper.router.compute_batch(OptimizerWrapper.config[:router][:url],
-                                                      vehicle[:router_mode].to_sym, vehicle[:router_dimension], segments, vrp.restitution_geometry_polyline, vehicle.router_options)
-      raise RouterError, 'Route details cannot be received' unless details
+                                                      vehicle.router_mode.to_sym, vehicle.router_dimension,
+                                                      segments, vrp.restitution_geometry.include?(:polyline),
+                                                      vehicle.router_options)
+      raise RouterError.new('Route details cannot be received') unless details
     end
+
     details&.each{ |d| d[0] = (d[0] / 1000.0).round(4) if d[0] }
     details
   end
 
-  def self.parse_result(vrp, result)
-    tic_parse_result = Time.now
-    result[:routes].each{ |r|
-      details = nil
-      v = vrp.vehicles.find{ |vehicle| vehicle.id == r[:vehicle_id] }
-      if r[:end_time] && r[:start_time]
-        r[:total_time] = r[:end_time] - r[:start_time]
-      end
+  def self.compute_route_travel_times(vrp, matrix, route, vehicle)
```

Comment:
'travel_times' implies to compute something relative to 'time' not a 'distance' ?

Reply:
Indeed..!